### ISSUE TITLE
fix(external-link): GET fallback does not trigger on HEAD 403/405

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#287]
## Problem

The `performCheck` function in `internal/rule/external_link.go` only falls back to GET when `client.Do` returns a **network error**. When HEAD succeeds at the transport level but returns a 403 or 405 status code, `err == nil` and the function returns immediately without retrying with GET.

```go
resp, err := client.Do(req)
if err == nil {
    defer resp.Body.Close()
    return resp.StatusCode, nil  // returns 403/405 — GET fallback never reached
}

// Only reached on network error, not on HTTP 4xx
req.Method = "GET"
```

PR #275 was titled "fall back to GET when HEAD returns 403 or 405" but the implementation does not match that description.

## Impact

Reported in #147: `https://paleoanthro.org/media/journal/content/PA20170044.pdf` is still flagged as unreachable on v3.0.6, even though the URL is valid and returns 200 on GET.

## Expected behaviour

When HEAD returns 403 or 405, `performCheck` should retry with GET before returning the status code.

## Fix

```go
resp, err := client.Do(req)
if err == nil {
    defer resp.Body.Close()
    if resp.StatusCode != http.StatusForbidden && resp.StatusCode != http.StatusMethodNotAllowed {
        return resp.StatusCode, nil
    }
    // fall through to GET fallback
}

req.Method = "GET"
```

**Comment by shinagawa-web** [comment]
Closing: this was filed in error. The fix in #275 is correctly implemented in main — I was reading from the wrong branch (ci/check-invisible-unicode) which predates the merge.
